### PR TITLE
New groovy output

### PR DESCRIPTION
**Pull request by adrbessy**
A Groovy Output that prints in the same way that the GroovyConsole.
There is a button allowing to delete its content.

### CODE REVIEW

File: plugins/org.orbisgis.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -19,6 +19,7 @@ Bundle-Activator: org.orbisgis.core.CoreActivator
 Bundle-ActivationPolicy: lazy
 Export-Package: org.orbisgis.core,
  org.orbisgis.core.logger,
+ org.orbisgis.core.service.definition,
```
Comment (ebocher) [inline plugins/org.orbisgis.core/META-INF/MANIFEST.MF:22]: to be removed ?

File: .github/workflows/CI-build.yml [hunk 2]
```diff
@@ -36,12 +36,12 @@ jobs:
...
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # GitHub will automatically add this - you don't need to bother getting a token
-          MESSAGE: "Build: plugins available from p2-repository" # The commit message
+      #- name: Push p2 build to another branch
+        #uses: s0/git-publish-subdir-action@master
+        #env:
+          #REPO: self
+          #BRANCH: p2-repository # The branch name where you want to push the assets
+          #FOLDER: ./p2/target/p2-1.0.0-SNAPSHOT # The directory where your assets are generated
+          #GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # GitHub will automatically add this - you don't need to bother getting a token
+          #MESSAGE: "Build: plugins available from p2-repository" # The commit message
```
Comment (ebocher) [inline .github/workflows/CI-build.yml:46]: disable ?

File: plugins/org.orbisgis.core/src/org/orbisgis/core/ui/ToolbarButton.java [hunk 3]
```diff
@@ -97,7 +105,7 @@ public void mouseHover(MouseEvent e) {
 
         this.addMouseListener(new MouseAdapter() {
             @Override
-            public void mouseDown(MouseEvent e) {
+            public void mouseDown(MouseEvent e) {
```
Comment (ebocher) [inline plugins/org.orbisgis.core/src/org/orbisgis/core/ui/ToolbarButton.java:108]: remove blank space.
You can used format with the ide

File: plugins/org.orbisgis.groovyeditor/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -21,4 +21,6 @@ Require-Bundle: org.eclipse.core.runtime,
  dependencies
 Bundle-ActivationPolicy: lazy
 Bundle-Activator: org.orbisgis.ui.editors.groovy.GroovyActivator
-
+Import-Package: org.jkiss.dbeaver.core,
+ org.jkiss.dbeaver.ext.sample.database,
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/META-INF/MANIFEST.MF:25]: necessary ?

File: plugins/org.orbisgis.groovyeditor/plugin.xml [hunk 5]
```diff
@@ -41,21 +54,40 @@
             description="Clear the script"
             id="org.orbisgis.ui.editors.groovy.clear"
             name="Clear script"> </command>
+<!--
+      <command
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/plugin.xml:58]: remove it

File: plugins/org.orbisgis.groovyeditor/plugin.xml [hunk 6]
```diff
@@ -68,8 +100,28 @@
             class="org.orbisgis.ui.editors.groovy.handlers.GroovyEditorActionHandler"
             commandId="org.orbisgis.ui.editors.groovy.clear">
       </handler>
+<!--
+      <handler
+            class="org.orbisgis.ui.editors.groovy.handlers.GroovyEditorActionHandler"
+            commandId="org.orbisgis.ui.editors.groovy.change.groovy.interpreter">
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/plugin.xml:106]: remove it

File: plugins/org.orbisgis.groovyeditor/plugin.xml [hunk 7]
```diff
@@ -102,6 +154,14 @@
             schemeId="org.eclipse.ui.defaultAcceleratorConfiguration"
             sequence="CTRL+DEL">
       </key>
+<!--
+      <key
+            commandId="org.orbisgis.ui.editors.groovy.change.groovy.interpreter"
+            contextId="org.orbisgis.ui.editors.groovy"
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/plugin.xml:160]: remove it

File: plugins/org.orbisgis.groovyeditor/pom.xml [hunk 8]
```diff
@@ -31,6 +31,10 @@
 			<artifactId>org.orbisgis.core</artifactId>
 			<version>1.0.0-SNAPSHOT</version>
 		</dependency>
+		<dependency>
+		    <groupId>org.osgi</groupId>
+		    <artifactId>org.osgi.core</artifactId>
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/pom.xml:36]: needed?

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/browser/OpenHandler.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+import org.eclipse.ui.browser.IWorkbenchBrowserSupport;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.ui.part.FileEditorInput;
+import org.orbisgis.core.CoreActivator;
+import org.orbisgis.core.logger.Logger;
+import org.orbisgis.core.workspace.ICoreWorkspace;
+import org.orbisgis.ui.editors.groovy.GroovyEditor;
+import org.orbisgis.ui.editors.groovy.GroovyUtils;
+
+public class OpenHandler extends AbstractHandler {
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/browser/OpenHandler.java:48]: Add author
Adrien Bessy, LAB-STICC, CNRS

Comment (ebocher) [reply plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/browser/OpenHandler.java:48]: This class is never used ?

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyConsoleView.java [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+	}
+	
+	@Override
+	public void setFocus() {
+		// TODO Auto-generated method stub
+	}
+	
+	private void createDeleteOutputButton(Composite group) {
+		final MenuManager menuMgr = new MenuManager();
+		menuMgr.setRemoveAllWhenShown(true);
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyConsoleView.java:53]: Format the code with the IDE

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyEditor.java [hunk 11]
```diff
@@ -140,4 +145,18 @@ public void stop(){
     public GroovyJob getRunningJob() {
         return job;
     }
+
+	public void changeGroovyInterpreter() {
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyEditor.java:149]: Never used ?

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyEditor.java [hunk 12]
```diff
@@ -140,4 +145,18 @@ public void stop(){
...
+		System.out.println("\n***\n in changeGroovyInterpreter method \n***\n");
+
+		if (groovyInterpreter == "GroovyShell") {
+			this.groovyInterpreter = "GroovyScriptEngine";
+			System.out.println("\n***\n groovyInterpreter in changeGroovyInterpreter method : " + groovyInterpreter + "\n***\n");
+			LOGGER.info("GroovyScriptEngine is now selected.");
+		} else {
+			this.groovyInterpreter = "GroovyShell";
+			System.out.println("\n***\n groovyInterpreter in changeGroovyInterpreter method : " + groovyInterpreter + "\n***\n");
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyEditor.java:158]: remove println

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyJob.java [hunk 13]
```diff
@@ -47,27 +59,32 @@
...
     private Thread t;
+    
 
-    public GroovyJob(String name, String script){
+    public GroovyJob(String name, String script, String groovyInterpreter){
         super(name);
         this.script = script;
         this.name = name;
-
+		System.out.println("\n***\n in GroovyJob constructor \n***\n");
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyJob.java:68]: Don't used any println in the code, use log system if necessary

File: plugins/org.orbisgis.groovyeditor/META-INF/MANIFEST.MF (hunk 4 above)
Comment (adrbessy) [reply plugins/org.orbisgis.groovyeditor/META-INF/MANIFEST.MF:25]: no!

File: plugins/org.orbisgis.groovyeditor/pom.xml (hunk 8 above)
Comment (adrbessy) [reply plugins/org.orbisgis.groovyeditor/pom.xml:36]: no!

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/browser/OpenHandler.java (hunk 9 above)
Comment (adrbessy) [reply plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/browser/OpenHandler.java:48]: This class allows to open a web page in the software, but we don't use it.

File: plugins/org.orbisgis.core/META-INF/MANIFEST.MF (hunk 1 above)
Comment (adrbessy) [reply plugins/org.orbisgis.core/META-INF/MANIFEST.MF]: Yes!

File: .github/workflows/CI-build.yml (hunk 2 above)
Comment (adrbessy) [reply .github/workflows/CI-build.yml]: The p2-1.0.0-SNAPSHOT cannot be pushed to p2-repository,
There is this message:
remote: Permission to orbisgis/orbisrcp.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/orbisgis/orbisrcp.git/': The requested URL returned error: 403

Comment (ebocher) [reply .github/workflows/CI-build.yml:46]: @SPalominos  could you check when you will have the time ;-)

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyJob.java [hunk 14]
```diff
@@ -128,15 +146,29 @@ public void run() {
...
                 else {
-                    result = shell.evaluate(script);
+                    for(String s : script.split("\n")) {
+                    	shell.evaluate(s);
+                    	if (s != null) {
+                    		GroovyConsoleContent.writeIntoConsole(s);
+                    	}
+                    }
                 }
+                Object res3 = shell.getProperty("out");
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyJob.java:156]: Format code

File: .github/workflows/CI-build.yml (hunk 2 above)
Comment (ebocher) [reply .github/workflows/CI-build.yml:46]: Is the GITHUB_TOKEN already active ?

File: plugins/org.orbisgis.groovyeditor/META-INF/MANIFEST.MF [hunk 15]
```diff
@@ -21,4 +21,4 @@ Require-Bundle: org.eclipse.core.runtime,
  dependencies
 Bundle-ActivationPolicy: lazy
 Bundle-Activator: org.orbisgis.ui.editors.groovy.GroovyActivator
-
+Import-Package: org.jkiss.dbeaver.registry.driver
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/META-INF/MANIFEST.MF:24]: needed ?

File: plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyJob.java [hunk 16]
```diff
@@ -47,27 +59,31 @@
     private Binding binding;
     private String name;
     private Thread t;
+    
 
-    public GroovyJob(String name, String script){
+    public GroovyJob(String name, String script, String groovyInterpreter){
```
Comment (ebocher) [inline plugins/org.orbisgis.groovyeditor/src/org/orbisgis/ui/editors/groovy/GroovyJob.java:64]: I don't under why you set a String argument to define the groovy engine.
I think you can remove it and use the original code. No ?

**Pull request closed without merging**